### ISSUE TITLE
Keycloak pods rolling update doesn't work

### ISSUE THREAD

**Issue by missedone** [keycloak/keycloak#29835]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

when we deployed the Keycloak cluster with the Kubernetes operator, the pod rolling update didn't work; all the old pods were deleted before the new pods were ready.
I didn't have the screenshot for now, but it happens sometimes.

### Version

24.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

spin up new pods and ready, then deleted on old pod, repeat the procedure until all new pods are up and old pods are deleted.

### Actual behavior

when we deployed the Keycloak cluster with the Kubernetes operator, the pod rolling update didn't work; all the old pods were deleted before the new pods were ready.
I didn't have the screenshot for now, but it happens sometimes.

### How to Reproduce?

1. have old version 24.0.3 running
2. deploy new version 24.0.4
3. we have the helm chart manage other the keycloak operator deployment and keycloak pods deployment, not sure if it matters.

**Comment by sschu** [comment]
@missedone Keycloak currently does not support updates without downtime because of potential database migrations and version incompatibilities in the embedded Infinispan. Therefore the update procedure needs to first stop all Keycloak pods before starting up new pods.

**Comment by daniel-ITC** [comment]
Hi @sschu,

Thank you for your clear response.

In my specific case, I have custom extensions that I build into a custom Docker image. As a result, I need to release a new version of the Docker image each time I want to update my extensions, but the Keycloak version remains unchanged.

I was considering implementing a blue/green deployment strategy for the Keycloak CRD. We use NGINX as our IngressClass. My plan was to deploy a new Keycloak CRD in the same namespace as the current one, naming the new instance something like keycloak-green, while pointing it to the same database and external Infinispan cluster. Once the new instance is ready, I would update the NGINX ingress to route traffic to the keycloak-green instance, followed by shutting down the old Keycloak CRD (referred to as blue).

Do you foresee any potential issues with this approach? My intention is not to use this for Keycloak version updates, but solely for updating the Docker image while maintaining the same Keycloak version.

**Comment by missedone** [comment]
@daniel-ITC , it seems that you have a different issue of the OP. you may raise a new ticket to discuss the details.

however, hope the below tips may solve your issue:
1. make sure you update the docker image tag, so the k8s API server can detect the change to trigger the rolling update.
2. if, somehow, you can't update the image tag, you can put the following snippet in your keycloak CR yaml to enforce the rolling update: 
```
      annotations:
        # used for triggering rolling update in case of new image with same tag
        deployTime: "{{ date "20060102150405" .Release.Time }}UTC"
```

**Comment by missedone** [comment]
reply to my original question, i ended up with migrating to custom helm chart (I think the [bitnami keycloak helm chart](https://artifacthub.io/packages/helm/bitnami/keycloak) should work too)

now the rolling update works as expected.

**Comment by sschu** [comment]
@daniel-ITC We have a similar setup. The way we do it is that we do a rolling update per default but if we detect a change in the Keycloak version, we scale Keycloak down before the update. It would be nice if the operator could also do this OOTB.
